### ISSUE TITLE
Class table inheritance plugin & auto validation plugin do not work together

### ISSUE THREAD

**Issue by rolftimmermans** [jeremyevans/sequel#1611]
### Complete Description of Issue

When using both the `auto_validations` plugin and the `class_table_inheritance` plugin, the auto validation does not work on any child models. No validations are run when attempting to save the child models.

### Simplest Possible Self-Contained Example Showing the Bug

```
require "sequel"

Sequel::Model.db = Sequel.postgres("test-db")

Sequel::Model.db.create_table("items") do
  primary_key :id
  text :kind, null: false
  text :value, null: false
end

Sequel::Model.db.create_table("child_items") do
  primary_key :id, table: :items
  text :child_value, null: false
end

Sequel::Model.class_eval do
  plugin(:auto_validations, not_null: :presence)
  self.raise_on_save_failure = false
end

class BaseItem < Sequel::Model(:items)
  plugin :class_table_inheritance, key: :kind,
    table_map: {ChildItem: :child_items},
    model_map: {"child" => :ChildItem}
end

class ChildItem < BaseItem
end

b = BaseItem.new(kind: "base")
p b.save
# => nil
p b.errors
# => {:value=>["is not present"]}

b = ChildItem.new
p b.save
# Raises:
# sequel-5.19.0/lib/sequel/adapters/postgres.rb:152:in `async_exec': PG::NotNullViolation: ERROR:  null value in column "value" violates not-null constraint (Sequel::NotNullConstraintViolation)
# DETAIL:  Failing row contains (8, child, null).
```

### Full Backtrace of Exception (if any)

```
Traceback (most recent call last):
	38: from foo.rb:37:in `<main>'
	37: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1465:in `save'
	36: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1896:in `checked_save_failure'
	35: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1465:in `block in save'
	34: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1905:in `checked_transaction'
	33: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/database/transactions.rb:186:in `transaction'
	32: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/database/connecting.rb:270:in `synchronize'
	31: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/connection_pool/threaded.rb:92:in `hold'
	30: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/database/transactions.rb:220:in `block in transaction'
	29: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/database/transactions.rb:245:in `_transaction'
	28: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1905:in `block in checked_transaction'
	27: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1465:in `block (2 levels) in save'
	26: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1745:in `_save'
	25: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1015:in `around_save'
	24: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1750:in `block in _save'
	23: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1015:in `around_create'
	22: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/model/base.rb:1753:in `block (2 levels) in _save'
	21: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/plugins/class_table_inheritance.rb:403:in `_insert'
	20: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/plugins/class_table_inheritance.rb:403:in `each'
	19: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/plugins/class_table_inheritance.rb:407:in `block in _insert'
	18: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/shared/postgres.rb:1557:in `insert_select'
	17: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/dataset/actions.rb:949:in `with_sql_first'
	16: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/dataset/actions.rb:941:in `with_sql_each'
	15: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:610:in `fetch_rows'
	14: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/dataset/actions.rb:1088:in `execute'
	13: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:314:in `execute'
	12: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/database/connecting.rb:270:in `synchronize'
	11: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/connection_pool/threaded.rb:88:in `hold'
	10: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:314:in `block in execute'
	 9: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:518:in `check_database_errors'
	 8: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:314:in `block (2 levels) in execute'
	 7: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:496:in `_execute'
	 6: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:140:in `execute'
	 5: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:116:in `check_disconnect_errors'
	 4: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:140:in `block in execute'
	 3: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:152:in `execute_query'
	 2: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/database/logging.rb:38:in `log_connection_yield'
	 1: from /Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:152:in `block in execute_query'
/Users/rolftimmermans/.gem/ruby/2.5.0/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:152:in `async_exec': PG::NotNullViolation: ERROR:  null value in column "value" violates not-null constraint (Sequel::NotNullConstraintViolation)
```

### SQL Log (if any)

```
LOG:  statement: SET standard_conforming_strings = ON
LOG:  statement: SET client_min_messages = 'WARNING'
LOG:  statement: SET DateStyle = 'ISO'
LOG:  statement: SELECT CAST(current_setting('server_version_num') AS integer) AS v
LOG:  statement: CREATE TABLE "items" ("id" serial PRIMARY KEY, "kind" text NOT NULL, "value" text NOT NULL)
DEBUG:  CREATE TABLE will create implicit sequence "items_id_seq" for serial column "items.id"
DEBUG:  building index "pg_toast_7150818_index" on table "pg_toast_7150818"
DEBUG:  CREATE TABLE / PRIMARY KEY will create implicit index "items_pkey" for table "items"
DEBUG:  building index "items_pkey" on table "items"
LOG:  statement: CREATE TABLE "child_items" ("id" serial PRIMARY KEY REFERENCES "items", "child_value" text NOT NULL)
DEBUG:  CREATE TABLE will create implicit sequence "child_items_id_seq" for serial column "child_items.id"
DEBUG:  building index "pg_toast_7150829_index" on table "pg_toast_7150829"
DEBUG:  CREATE TABLE / PRIMARY KEY will create implicit index "child_items_pkey" for table "child_items"
DEBUG:  building index "child_items_pkey" on table "child_items"
LOG:  statement: SELECT "pg_attribute"."attname" AS "name", CAST("pg_attribute"."atttypid" AS integer) AS "oid", CAST("basetype"."oid" AS integer) AS "base_oid", format_type("basetype"."oid", "pg_type"."typtypmod") AS "db_base_type", format_type("pg_type"."oid", "pg_attribute"."atttypmod") AS "db_type", pg_get_expr("pg_attrdef"."adbin", "pg_class"."oid") AS "default", NOT "pg_attribute"."attnotnull" AS "allow_null", COALESCE(("pg_attribute"."attnum" = ANY("pg_index"."indkey")), false) AS "primary_key" FROM "pg_class" INNER JOIN "pg_attribute" ON ("pg_attribute"."attrelid" = "pg_class"."oid") INNER JOIN "pg_type" ON ("pg_type"."oid" = "pg_attribute"."atttypid") LEFT OUTER JOIN "pg_type" AS "basetype" ON ("basetype"."oid" = "pg_type"."typbasetype") LEFT OUTER JOIN "pg_attrdef" ON (("pg_attrdef"."adrelid" = "pg_class"."oid") AND ("pg_attrdef"."adnum" = "pg_attribute"."attnum")) LEFT OUTER JOIN "pg_index" ON (("pg_index"."indrelid" = "pg_class"."oid") AND ("pg_index"."indisprimary" IS TRUE)) WHERE (("pg_attribute"."attisdropped" IS FALSE) AND ("pg_attribute"."attnum" > 0) AND ("pg_class"."oid" = CAST(CAST('"items"' AS regclass) AS oid))) ORDER BY "pg_attribute"."attnum"
LOG:  statement: SELECT "indc"."relname" AS "name", "ind"."indisunique" AS "unique", "att"."attname" AS "column", "con"."condeferrable" AS "deferrable" FROM "pg_class" AS "tab" INNER JOIN "pg_index" AS "ind" ON ("ind"."indrelid" = "tab"."oid") INNER JOIN "pg_class" AS "indc" ON ("indc"."oid" = "ind"."indexrelid") INNER JOIN "pg_attribute" AS "att" ON (("att"."attrelid" = "tab"."oid") AND ("att"."attnum" = ANY("ind"."indkey"))) LEFT JOIN "pg_constraint" AS "con" ON ("con"."conname" = "indc"."relname") WHERE (("indc"."relkind" = 'i') AND ("ind"."indisprimary" IS FALSE) AND ("indexprs" IS NULL) AND ("indisvalid" IS TRUE) AND ("tab"."oid" = CAST(CAST('"items"' AS regclass) AS oid)) AND ("indpred" IS NULL) AND ("indisready" IS TRUE) AND ("indislive" IS TRUE)) ORDER BY "indc"."relname", array_position("ind"."indkey", "att"."attnum")
LOG:  statement: SELECT "indc"."relname" AS "name", "ind"."indisunique" AS "unique", "att"."attname" AS "column", "con"."condeferrable" AS "deferrable" FROM "pg_class" AS "tab" INNER JOIN "pg_index" AS "ind" ON ("ind"."indrelid" = "tab"."oid") INNER JOIN "pg_class" AS "indc" ON ("indc"."oid" = "ind"."indexrelid") INNER JOIN "pg_attribute" AS "att" ON (("att"."attrelid" = "tab"."oid") AND ("att"."attnum" = ANY("ind"."indkey"))) LEFT JOIN "pg_constraint" AS "con" ON ("con"."conname" = "indc"."relname") WHERE (("indc"."relkind" = 'i') AND ("ind"."indisprimary" IS FALSE) AND ("indexprs" IS NULL) AND ("indisvalid" IS TRUE) AND ("tab"."oid" = CAST(CAST('"items"' AS regclass) AS oid)) AND ("indpred" IS NULL) AND ("indisready" IS TRUE) AND ("indislive" IS TRUE)) ORDER BY "indc"."relname", array_position("ind"."indkey", "att"."attnum")
LOG:  statement: SELECT "indc"."relname" AS "name", "ind"."indisunique" AS "unique", "att"."attname" AS "column", "con"."condeferrable" AS "deferrable" FROM "pg_class" AS "tab" INNER JOIN "pg_index" AS "ind" ON ("ind"."indrelid" = "tab"."oid") INNER JOIN "pg_class" AS "indc" ON ("indc"."oid" = "ind"."indexrelid") INNER JOIN "pg_attribute" AS "att" ON (("att"."attrelid" = "tab"."oid") AND ("att"."attnum" = ANY("ind"."indkey"))) LEFT JOIN "pg_constraint" AS "con" ON ("con"."conname" = "indc"."relname") WHERE (("indc"."relkind" = 'i') AND ("ind"."indisprimary" IS FALSE) AND ("indexprs" IS NULL) AND ("indisvalid" IS TRUE) AND ("tab"."oid" = CAST(CAST('"items"' AS regclass) AS oid)) AND ("indpred" IS NULL) AND ("indisready" IS TRUE) AND ("indislive" IS TRUE)) ORDER BY "indc"."relname", array_position("ind"."indkey", "att"."attnum")
LOG:  statement: SELECT "pg_attribute"."attname" AS "name", CAST("pg_attribute"."atttypid" AS integer) AS "oid", CAST("basetype"."oid" AS integer) AS "base_oid", format_type("basetype"."oid", "pg_type"."typtypmod") AS "db_base_type", format_type("pg_type"."oid", "pg_attribute"."atttypmod") AS "db_type", pg_get_expr("pg_attrdef"."adbin", "pg_class"."oid") AS "default", NOT "pg_attribute"."attnotnull" AS "allow_null", COALESCE(("pg_attribute"."attnum" = ANY("pg_index"."indkey")), false) AS "primary_key" FROM "pg_class" INNER JOIN "pg_attribute" ON ("pg_attribute"."attrelid" = "pg_class"."oid") INNER JOIN "pg_type" ON ("pg_type"."oid" = "pg_attribute"."atttypid") LEFT OUTER JOIN "pg_type" AS "basetype" ON ("basetype"."oid" = "pg_type"."typbasetype") LEFT OUTER JOIN "pg_attrdef" ON (("pg_attrdef"."adrelid" = "pg_class"."oid") AND ("pg_attrdef"."adnum" = "pg_attribute"."attnum")) LEFT OUTER JOIN "pg_index" ON (("pg_index"."indrelid" = "pg_class"."oid") AND ("pg_index"."indisprimary" IS TRUE)) WHERE (("pg_attribute"."attisdropped" IS FALSE) AND ("pg_attribute"."attnum" > 0) AND ("pg_class"."oid" = CAST(CAST('"child_items"' AS regclass) AS oid))) ORDER BY "pg_attribute"."attnum"
LOG:  statement: BEGIN
LOG:  statement: INSERT INTO "items" ("kind") VALUES ('child') RETURNING *
ERROR:  null value in column "value" violates not-null constraint
DETAIL:  Failing row contains (1, child, null).
STATEMENT:  INSERT INTO "items" ("kind") VALUES ('child') RETURNING *
LOG:  statement: ROLLBACK
```

**Comment by jeremyevans** [comment]
Thanks for the report. I think it should be possible to get non-unique validations to work at least, but some basic things I've tried haven't fixed it yet.  I'm currently at a conference, but I will work on this and try to fix it before the next release.

**Comment by rolftimmermans** [comment]
Excellent! This fix does seem to work for columns from the parent table (like in my example above), but not for columns from the child table. Would you like me to come up with another test example or does it make sense like this?

**Comment by jeremyevans** [comment]
No need to come up with another example, I see what you are saying and I agree that this is only a partial fix.  I have an idea for a better fix, let me see if it will work.

**Comment by jeremyevans** [comment]
This should be fixed by 40d00b287cd7a99bcf812bc414562b57cabf4436.
